assessments/projects/fluentd/fluentd/self-assessment.md

```diff
@@ -0,0 +1,194 @@
+
+![Fluentd Sources and Destinations](https://www.fluentd.org/images/blog/unified-logging-pluggable.png)
+
+Fluentd is a relatively lightweight program in its vanilla instance, and there are 3 main actors in the architecture of Fluentd:
+- **Core Configuration**: The configuration file sets up rules and parameters for logging events by specifying plugins. The Fluentd core executable initializes that pipeline of plugins, and runs concurrently with the plugins to continually process events.
+- **Plugins**: Fluentd is composed of plugins that perform specific functions. Each plugin is executed independently of the core process and of each other, but exchange data along the pipeline.
+- **Plugin API**: Fluentd has Plugin APIs for third parties to create custom plugins. There are 2 versions for the current v1 and the old v0.12. Fluentd v1 supports both v1 and v0.12 APIs.
+
+### Actions
+Fluentd log pipelines consist of different types of plugins stitched together to perform specific and customizable data processing operations. Some of these plugins are available in the base package, while others are installed on Fluentd. There are over 500 community plugins for Fluentd that can read data including enterprise staples like Kafka, as well as data heavy backends like Hadoop and MongoDB. Generally speaking, a Fluentd pipeline consists of three stages, input, processing, and output, which are setup by configuring different Fluentd plugins.
+
+There are a total of 9 types of plugins in Fluentd: input, parser, filter, output, formatter, storage, service discovery, buffer, and metrics. The configuration file allows the user to define inputs or listeners and set up common matching rules to route logging event data to specific outputs by specifying and configuring these plugins. Each logging event is processed by the core configuration, which initializes the plugins in the order in which they are specified in the configuration file. The events follow a step-by-step cycle where they are processed in order, from top to bottom, and the plugins send logging data down the pipeline. Fluentd is equipped with a password-based authentication mechanism, which verifies the identity of each client using a shared secret key. For example, the secure_forward plugins for input and output (in_secure_forward and out_secure_forward) accepts and sends messages via SSL with authentication. Fluentd has a flexible plugin system that allows the community to extend its functionality, but any Fluentd plugin can unknowingly break Fluentd completely (and possibly break other plugins) by requiring some incompatible modules.
```

Comment:
I like your analysis at a high level, but maybe breaking out the plugin types or the general operation capabilities they provide into sub-sections of the actions section could be helpful for flow and comprehension.

Reply:
Hi @torinvdb, thank you for your feedback. I believe the reason its not further into detail is because we created a self assessment folder section called plugin where more information regarding the plugins would be given as suggested by another review earlier in the week.

Reply:
Also, a previous reviewer told us to take out information regarding the plugins and we made this change regarding a more basic overview based on their suggestion. You may scroll to the previously resolved comments to view this.

Reply:
That makes more sense, thank you!